components/cdc/src/delegate.rs

```diff
@@ -362,13 +362,12 @@ impl Delegate {
             self.region_id,
             change_data_event,
         );
-        for i in 0..downstreams.len() - 1 {
-            if normal_only && downstreams[i].state.load() != DownstreamState::Normal {
+        for downstream in downstreams {
```

Comment:
I don't think they are the same. The original implement will always choose the last downstream if all downstreas are not normal.

Reply:
@5kbpers said it was a bug on master.